### PR TITLE
[MIG] 14.0 report_context

### PR DESCRIPTION
**Pull request by PierrickBrun**

### CODE REVIEW

**Comment by etobella** [comment]
Printing of reports is not working:

imagen

File: report_context/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+from odoo.http import request, route
+
+from odoo.addons.web.controllers import main as report
+
+
+class ReportController(report.ReportController):
+    @route()
+    def report_routes(self, reportname, docids=None, converter=None, **data):
+        report = request.env["ir.actions.report"]._get_report_from_name(reportname)
+        original_context = json.loads(data.get("context", "{}"))
```
Comment (petrus-v) [inline report_context/controllers/main.py]: > Printing of reports is not working:
> 
> imagen

I've investigated a bit, I think there are two things to consider:

* [ ] We can get data likes this `{"context": None}` so I suggest:
```suggestion
        original_context = json.loads(data.get("context", "{}") or "{}")
```

* [ ] according the dependency tree and current traceback we could get data with context as dict (not str) if report_xlsx is installed; `{"context": {}}`

not sure what's the proper way to achive that maybe fix that in report_xlix to dump that context: https://github.com/OCA/reporting-engine/blob/14.0/report_xlsx/controllers/main.py#L26 or manage some optional dependency to ensure this method is called before if report_xlsx is installed as well (not sure this exist in odoo?).

**Comment by sebastienbeau** [comment]
@PierrickBrun can you check  this ?

**Comment by PierrickBrun** [comment]
I have no problem with printing on my local install.

I can't print from runbot but it does not matter if the module is installed or not.

@etobella how did you get this error ?

File: report_context/controllers/main.py (hunk 1 above)
Comment (petrus-v) [reply report_context/controllers/main.py]: sorry I've written this few weeks ago and didn't realize I've forgot to submit it, I'm not sure if it's still true this was just before @sebastienbeau comment !

Comment (PierrickBrun) [reply report_context/controllers/main.py]: > sorry I've written this few weeks ago and didn't realize I've forgot to submit it, I'm not sure if it's still true this was just before @sebastienbeau comment !

No problem, thank you for the investigation

**Comment by etobella** [comment]
@PierrickBrun Can you rebase it?

**Comment by PierrickBrun** [comment]
@etobella Done,

CI won't run but it looks like an unrelated problem:

`http://security.ubuntu.com/ubuntu/pool/main/o/openjdk-lts/openjdk-11-jre-headless_11.0.14.1+1-0ubuntu1~20.04_amd64.deb  404  Not Found [IP: 185.125.190.36 80]`

It would need to be rebased again in the future

**Comment by etobella** [comment]
Can you rebase and execute again? The issue was solved yesterday

**Comment by PierrickBrun** [comment]
It works now @etobella

File: report_context/readme/USAGE.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
+To configure this module, you need to:
+
+* Enter Odoo in debug mode.
+* To add a specific context to a report, you should go to Settings ->
+  Reporting -> Reporting and look for the report you want to edit on the
```
Comment (olgamarcocb) [inline report_context/readme/USAGE.rst]: Reporting -> Reports

**Review by olgamarcocb — APPROVED** [review]
LGTM 👍

**Review by sebastienbeau — APPROVED** [review]
LGTM code review and test

**Pull request merged**